Microsoft.Azure.CosmosRepository/src/InMemoryRepository.cs

```diff
@@ -134,6 +134,14 @@ public async ValueTask DeleteAsync(string id, PartitionKey partitionKey, Cancell
             Items.TryRemove(item!.Id, out _);
         }
 
+        /// <inheritdoc/>
+        public async ValueTask<int> CountAsync(Expression<Func<TItem, bool>> predicate,
+            CancellationToken cancellationToken = default)
+        {
+            await Task.CompletedTask;
+            return Items.Values.Where(predicate.Compose(
+                item => item.Type == typeof(TItem).Name, Expression.AndAlso).Compile()).Count();
+        }
```

Comment:
```suggestion
        public async ValueTask<int> CountAsync(Expression<Func<TItem, bool>> predicate,
            CancellationToken cancellationToken = default) =>
            await Task.FromResult(
                Items.Values.Count(predicate.Compose(
                    item => item.Type == typeof(TItem).Name, Expression.AndAlso).Compile()));

```

Added space after method, prefer expression-bodied member, use `FromResult` instead of `CompletedTask`, and `Count` instead of `Where(...).Count`. I think that will work, but some tests would be GREAT.